RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -444,19 +438,27 @@ PixelCPEClusterRepair::callTempReco3D( DetParam const & theDetParam,
    float deltay = 0;    // return param
    int npixels = 0;     // return param
 
-   theClusterParam.ierr2 =
-   PixelTempReco3D( ID, theClusterParam.cotalpha, theClusterParam.cotbeta,
-                   locBz, locBx,
-		   edgeTypeY , theClusterParam.edgeTypeX_ ,
-                   clusterPayload,
-                   templ2d,
-                   theClusterParam.templYrec_, theClusterParam.templSigmaY_, 
-                   theClusterParam.templXrec_, theClusterParam.templSigmaX_, 
-		   theClusterParam.templProbXY_,
-         	   theClusterParam.probabilityQ_,
-		   theClusterParam.qBin_,
-		   deltay, npixels
-                   );
+   if(clusterPayload.mrow > 4){
+       // The cluster is too big, the 2D reco will perform horribly.
+       // Better to return immediately in error
+       theClusterParam.ierr = 8;
```

Comment:
```theClusterParam.ierr2``` , maybe?

Reply:
I concur, ierr2 would make more sense since this replaces the second template fit.  Thanks for catching this.